scipy/sparse/linalg/propack.py

```diff
@@ -0,0 +1,329 @@
+    # of 16, but docs don't specify; it's almost surely a
+    # power of 2.
+    blocksize = 16
+
+    # Determine lwork & liwork:
+    # the required lengths are specified in the PROPACK documentation
+    if compute_u or compute_v:
+        lwork = m + n + 9*kmax + 5*kmax*kmax + 4 + max(3*kmax*kmax + 4*kmax + 4,
+                                                       blocksize*max(m, n))
+        liwork = 8*kmax
+    else:
+        lwork = m + n + 9*kmax + 2*kmax*kmax + 4 + max(m + n, 4*kmax + 4)
+        liwork = 2*kmax + 1
+    work = np.empty(lwork, dtype=typ.lower())
+    iwork = np.empty(liwork, dtype=np.int32)
+
+    # dummy arguments: these are passed to aprod, and not used in this wrapper
+    dparm = np.empty(1, dtype=typ.lower())
+    iparm = np.empty(1, dtype=np.int32)
+
+    if typ.isupper():
+        # PROPACK documentation is unclear on the required length of zwork.
+        # Use the same length Julia's wrapper uses
+        # see https://github.com/JuliaSmoothOptimizers/PROPACK.jl/
+        zwork = np.empty(m + n + 32*m, dtype=typ)
+
+        if irl_mode:
+            u, sigma, bnd, v, info = lansvd_irl(_which_converter[which],
+                                                jobu, jobv, m, n,
+                                                shifts, k, maxiter, aprod,
+                                                u, v, tol, work, zwork,
+                                                iwork, doption, ioption,
+                                                dparm, iparm)
+        else:
+            u, sigma, bnd, v, info = lansvd(jobu, jobv, m, n, k, aprod, u, v,
+                                            tol, work, zwork, iwork, doption,
+                                            ioption, dparm, iparm)
+    else:
+        if irl_mode:
+            u, sigma, bnd, v, info = lansvd_irl(_which_converter[which],
+                                                jobu, jobv, m, n,
+                                                shifts, k, maxiter, aprod,
+                                                u, v, tol, work, iwork,
+                                                doption, ioption, dparm,
+                                                iparm)
+        else:
+            u, sigma, bnd, v, info = lansvd(jobu, jobv, m, n, k, aprod, u, v,
+                                            tol, work, iwork, doption,
+                                            ioption, dparm, iparm)
+
+    if info > 0:
+        raise LinAlgError(
+            f"An invariant subspace of dimension {info} was found.")
+    elif info < 0:
+        raise LinAlgError(
+            f"k={k} singular triplets did not converge within "
+            f"kmax={kmax} iterations")
+
+    # info == 0: The K largest (or smallest) singular triplets were computed
+    # succesfully!
+
+    # catch corner case where A is all zeros and v is not orthogonal
+    if not np.any(v):
+        v = np.eye(*v.shape)
```

Comment:
I tested this out - but in this case, `u` is still not orthogonal. 
I'd suggest removing this corner case treatment because it's part of a larger issue that also occurs when `info != 0`. (When `info > 0`, the singular vectors after those corresponding with the first `info` singular values are not orthogonal.)

Not sure what to do here. Does PROPACK have any routines for doing e.g. Gram-Schmidt just to get orthogonal vectors? Maybe `_augmented_orthonormal_cols/rows` could help? 

Personally, I don't really want to dig into it, though. I think it would be nice if we could just say that singular vectors corresponding with zero singular values are not necessarily orthogonal.

Reply:
So the suggestion is to remove these 3 lines?  I'm all for it -- this is a strange corner case and I can't imagine a user that needs orthogonal `u`, `vh` in this case.  Of course I could be wrong, but I agree replicating PROPACK's real behavior is the most straightforward way to handle this

Reply:
Yes, I'm suggesting that we remove the three lines because they fix a very small part of a much larger problem.
I do think we should fix it more completely, eventually, but for now I think it's ok to just return PROPACK's results.

Reply:
After #21 is merged I'll remove these lines

Reply:
Removed